### PR TITLE
Added new move methods

### PR DESCRIPTION
**Pull request by AlexanderBlackman**
I've added three new methods and put an example of each into example app.

Nothing has been deleted.

### CODE REVIEW

File: src/WinUIExSample/MainWindow.xaml [hunk 1]
```diff
@@ -2,16 +2,18 @@
...
-    Width="1024" Height="768"
-    MinWidth="500" MinHeight="250"
+    Width="1024"
+    Height="768"
+    MinWidth="500"
+    MinHeight="250"
+    TaskBarIcon="Images/WindowIcon.ico"
     mc:Ignorable="d">
-    
+
```
Comment (dotMorten) [inline src/WinUIExSample/MainWindow.xaml:16]: These changes seem unrelated?

File: src/WinUIExSample/MainWindow.xaml [hunk 2]
```diff
@@ -20,47 +22,145 @@
         <TextBlock Text="WinUI Window Extensions" VerticalAlignment="Center" />
     </winex:WindowEx.TitleBar>-->
 
-    <Grid ColumnDefinitions="Auto,Auto,*" Margin="20">
+    <Grid Margin="20" ColumnDefinitions="Auto,Auto,*">
```
Comment (dotMorten) [inline src/WinUIExSample/MainWindow.xaml:25]: Unrelated change.

File: src/WinUIExSample/MainWindow.xaml [hunk 3]
```diff
@@ -20,47 +22,145 @@
...
+                        Value="300" 
+                        Margin="5"/>
+                </StackPanel>
+                <StackPanel Orientation="Horizontal" HorizontalAlignment="Center">
+                <Button Click="MoveToPoint_Click" Content="Move to XY" />
+                <Button Click="CentreOnPoint_Click" Content="Centre on XY" />                    
+                </StackPanel>
+
+            </StackPanel>
+        </ScrollViewer>
```
Comment (dotMorten) [inline src/WinUIExSample/MainWindow.xaml:152]: In general don't make PRs that makes unrelated changes - it makes it near impossible to review what you actually changes, and all these reordering and reformatting changes aren't necessary and might not match the maintainer's preference. Could you please undo all those so only the things you added show up?

File: src/WinUIExSample/MainWindow.xaml [hunk 4]
```diff
@@ -20,47 +22,145 @@
...
+        <Grid Grid.Column="2" RowDefinitions="Auto,*">
             <TextBlock Text="Log" />
-            <TextBox IsReadOnly="True" AcceptsReturn="True" VerticalAlignment="Stretch" HorizontalAlignment="Stretch" x:Name="WindowEventLog" Grid.Row="1" />
+            <TextBox
+                x:Name="WindowEventLog"
+                Grid.Row="1"
+                HorizontalAlignment="Stretch"
+                VerticalAlignment="Stretch"
+                AcceptsReturn="True"
+                IsReadOnly="True" />
```
Comment (dotMorten) [inline src/WinUIExSample/MainWindow.xaml:162]: Undo

File: src/WinUIExSample/MainWindow.xaml [hunk 5]
```diff
@@ -20,47 +22,145 @@
...
+                        Grid.Column="1"
+                        Click="moveWindowButton_Clicked"
+                        Content="BottomCenter" />
+                    <Button
+                        Grid.Row="2"
+                        Grid.Column="2"
+                        Click="moveWindowButton_Clicked"
+                        Content="BottomRight" />
+
+                </Grid>
```
Comment (dotMorten) [inline src/WinUIExSample/MainWindow.xaml:129]: Might be cleaner/simpler with a drop down?

File: src/WinUIExSample/MainWindow.xaml [hunk 6]
```diff
@@ -20,47 +22,145 @@
...
+                <RadioButton Content="TopRight"  Checked="moveWindowRadioButtons_Checked"/>
+
+                <RadioButton Content="MiddleLeft"  Checked="moveWindowRadioButtons_Checked"/>
+                <RadioButton Content="MiddleCenter"  Checked="moveWindowRadioButtons_Checked"/>
+                <RadioButton Content="MiddleRight"   Checked="moveWindowRadioButtons_Checked"/>
+
+                <RadioButton Content="BottomLeft"  Checked="moveWindowRadioButtons_Checked"/>
+                <RadioButton Content="BottomCenter"  Checked="moveWindowRadioButtons_Checked"/>
+                <RadioButton Content="BottomRight" Checked="moveWindowRadioButtons_Checked"/>
+                </RadioButtons>-->
```
Comment (dotMorten) [inline src/WinUIExSample/MainWindow.xaml:86]: Dead code

File: src/WinUIExSample/MainWindow.xaml.cs [hunk 7]
```diff
@@ -170,5 +170,30 @@ private void WindowMessageReceived(object sender, WindowMessageEventArgs e)
...
+        }
+
+        //private void moveWindowRadioButtons_Checked(object sender, RoutedEventArgs e)
+        //{
+        //    var rb = sender as RadioButton;
+        //    HwndExtensions.Positions position = (HwndExtensions.Positions)Enum.Parse(typeof(HwndExtensions.Positions), rb.Content.ToString());          
+        //    this.MoveToSelectedPosition(position);
+
+        //}
+
```
Comment (dotMorten) [inline src/WinUIExSample/MainWindow.xaml.cs:191]: Dead code

File: src/WinUIEx/HwndExtensions.cs [hunk 8]
```diff
@@ -115,6 +115,131 @@ public static void CenterOnScreen(IntPtr hwnd, double? width = null, double? hei
...
+            TopCenter,
+            TopRight,
+            MiddleLeft,
+            MiddleCenter,
+            MiddleRight,
+            BottomLeft,
+            BottomCenter,
+            BottomRight
+
+        }
```
Comment (dotMorten) [inline src/WinUIEx/HwndExtensions.cs:171]: This should probably be in the WinUIEx namespace instead of inside a class - especially considering it is used in multiple places and not just in this class. Also this should generate warnings as there is no summary on each value.
Lastly can we come up with a better name for this enum? (also shouldn't be plural)

File: src/WinUIEx/HwndExtensions.cs [hunk 9]
```diff
@@ -115,6 +115,131 @@ public static void CenterOnScreen(IntPtr hwnd, double? width = null, double? hei
...
+
+        /// <summary>
+        /// Moves the window to the specified enum edge position on the current monitor.
```
Comment (dotMorten) [inline src/WinUIEx/HwndExtensions.cs:174]: Clever idea. Do we need an overload for specifying another monitor? Also would an optional offset make sense? In a way that makes MoveToPosition irrelevant. In fact we could probably merge all 3 methods into one with a few overloads, since they all do exactly the same thing: 
```cs
MoveTo(IntPtr handle, double x, double y) => MoveTo(handle, x, y, Positions.TopLeft);
MoveTo(IntPtr handle, Positions edge) => MoveTo(handle, 0, 0, edge);
MoveTo(IntPtr handle, double x, double y, Positions relativeTo) { ... }
```

File: src/WinUIEx/HwndExtensions.cs [hunk 10]
```diff
@@ -115,6 +115,131 @@ public static void CenterOnScreen(IntPtr hwnd, double? width = null, double? hei
...
+        /// <summary>
+        /// The edge positions MoveToPosition can move to
+        /// </summary>
+        public enum Positions
+        {
+            TopLeft,
+            TopCenter,
+            TopRight,
+            MiddleLeft,
+            MiddleCenter,
```
Comment (dotMorten) [inline src/WinUIEx/HwndExtensions.cs:165]: Middle and Center makes an odd combo :-)
Which makes me wonder if we should just reuse Horizontal and Vertical Alignment enums?

**Review by dotMorten — CHANGES_REQUESTED** [review]
Great idea! Some suggestions for improvement to the API, plus a request to undo all the unrelated changes to XAML.

File: src/WinUIEx/HwndExtensions.cs (hunk 8 above)
Comment (AlexanderBlackman) [reply src/WinUIEx/HwndExtensions.cs:171]: Good idea, I'll move it.
WindowPosition? Position?

File: src/WinUIEx/HwndExtensions.cs (hunk 10 above)
Comment (AlexanderBlackman) [reply src/WinUIEx/HwndExtensions.cs:165]: I based it on the ContentAlignment Enum in System.Drawing.

File: src/WinUIExSample/MainWindow.xaml.cs (hunk 7 above)
Comment (AlexanderBlackman) [reply src/WinUIExSample/MainWindow.xaml.cs:191]: Understood, I didn't know if you'd prefer radio buttons to buttons, but I'll delete it.

File: src/WinUIExSample/MainWindow.xaml (hunk 5 above)
Comment (AlexanderBlackman) [reply src/WinUIExSample/MainWindow.xaml:129]: Not in this case, as the buttons are positioned where they will move the window to.

File: src/WinUIEx/HwndExtensions.cs (hunk 8 above)
Comment (dotMorten) [reply src/WinUIEx/HwndExtensions.cs:171]: How about just “Alignment” or “Placement”

File: src/WinUIExSample/MainWindow.xaml (hunk 3 above)
Comment (AlexanderBlackman) [reply src/WinUIExSample/MainWindow.xaml:152]: Will do.

File: src/WinUIExSample/MainWindow.xaml (hunk 1 above)
Comment (AlexanderBlackman) [reply src/WinUIExSample/MainWindow.xaml:16]: My bad, I forgot I was using the XAML formatter extension.  This is my first time contributing to a open source project.

Comment (dotMorten) [reply src/WinUIExSample/MainWindow.xaml:16]: No worries. You're doing awesome! Keep it up.
As a future tip, when you go to submit a new PR, you should see all the changes below. Always good to scroll over it and make sure only the changes you really need are in there before you hit that submit button. Formatting and cleanup of unrelated code just makes it a lot harder on the maintainers.

File: src/WinUIExSample/MainWindow.xaml (hunk 5 above)
Comment (dotMorten) [reply src/WinUIExSample/MainWindow.xaml:129]: Aaah I totally missed that. Love it!

File: src/WinUIEx/HwndExtensions.cs (hunk 9 above)
Comment (AlexanderBlackman) [reply src/WinUIEx/HwndExtensions.cs:174]: Good idea,  I'll do it right now.
As for multiple monitors, I'll look into it before making any changes.
